### ISSUE TITLE
Implementing an event manager

### ISSUE THREAD

**Issue by ITPini** [PrometheusBork/AI-RTS-PROJECT#17]
Implementing an event manager for inputs might be a reasonable time to do so
```python
if event.key == pygame.K_i:
    if self.selection_manager.get_selected_object() == self.game_world.player_manager.players[0].base:
        self.handle_unit_creation(self.game_world.player_manager.players[0], InfantryUnit(), 50)
    elif self.selection_manager.get_selected_object() == self.game_world.player_manager.players[1].base:
        self.handle_unit_creation(self.game_world.player_manager.players[1], InfantryUnit(), 50)
if event.key == pygame.K_w:
    if self.selection_manager.get_selected_object() == self.game_world.player_manager.players[0].base:
        self.handle_unit_creation(self.game_world.player_manager.players[0], WorkerUnit(), 25)
    elif self.selection_manager.get_selected_object() == self.game_world.player_manager.players[1].base:
        self.handle_unit_creation(self.game_world.player_manager.players[1], WorkerUnit(), 25)
```

_Originally posted by @ITPini in https://github.com/PrometheusBork/AI-RTS-PROJECT/pull/15#discussion_r1550043611_

**Comment by PrometheusBork** [comment]
I have made a branch for this issue: enhancement/event-manager

**Comment by PrometheusBork** [comment]
resolved in #19
